src/Application.cpp

```diff
@@ -18,6 +18,32 @@
 Application::Application(int& argc, char** argv) :
     QApplication(argc, argv)
 {
+    // Check 'DB4S_CONFIG_FILE environment variable exists
+    const char* env = getenv("DB4S_CONFIG_FILE");
+
+    // If 'DB4S_CONFIG_FILE' environment variable exists
+    if(env)
+    {
+        Settings::userConfiguration = true;
+        Settings::userConfigurationDir = QString::fromStdString(env);
```

Comment:
This would be more elegant with a setter and only the configuration file is needed. If it's empty, no configuration file has been set.